lib/live-collection.js

```diff
@@ -191,10 +198,33 @@ module.exports = Backbone.Collection.extend({
       this.reset(snapshot, options);
     }
 
+    this._snapshotReceived = true;
+
     this.trigger('sync');
     this.trigger('snapshot');
   },
 
+  onSnapshotReceived: function() {
+    var self = this;
+    if (this._snapshotReceived) {
+      debug('Snapshot already received, resolving immediately');
+      return Promise.resolve();
+    } else {
+      debug('Awaiting snapshot');
+      return new Promise(function(resolve, reject) {
+        self.once('snapshot', function() {
+          var resolved = true;
+          debug('Snapshot received');
+          resolve();
+        });
+        self.once('subscriptionError', function() {
```

Comment:
You still need to forward the event from the templateSubscription. Maybe a test would help?